source/slang/glsl.meta.slang

```diff
@@ -9601,3 +9601,14 @@ public bool allInvocationsEqual(bool value)
 {
     return WaveActiveAllEqual(value);
 }
+
+
+/// extensions
+/// https://github.com/KhronosGroup/GLSL/blob/main/extensions/ext/GL_EXT_nonuniform_qualifier.txt
+
+__generic<T:__BuiltinArithmeticType>
+__intrinsic_op($(kIROp_NonUniformResourceIndex))
+[require(cpp_cuda_glsl_hlsl_spirv, nonuniformqualifier)]
```

Comment:
"cpp_cuda_glsl_hlsl_spirv" seems surprising. Isn't this a GLSL-only thing?

Reply:
it's implemented by the slang IR op `kIROp_NonUniformResourceIndex`, which is supported on those platforms. The requirement here is for the backend, not the source language. It should be exactly the same implementation as HLSL's `NonUniformResourceIndex`

Reply:
I am piggy backing on how HLSL implemented.

Alternative implementation could be something like,
```
#define nonuniformEXT NonUniformResourceIndex
```
But this would be ugly.

Reply:
I see, thanks for the explanation!